src/components/chat/commented-documents.tsx

```diff
@@ -165,17 +165,20 @@ export const CommentedDocuments: React.FC<IProps> = ({user, handleDocView}) => {
       {
         workDocuments &&
         (workDocuments).map((doc: PromisedDocumentDocument, index: number) =>{
+          //"Student Workspaces/"My Work"/"Class Work"
           const sectionDoc =  store.documents.getDocument(doc.key);
           const networkDoc = store.networkDocuments.getDocument(doc.key);
+          const isStudentWorkspaceDoc = (user?.id !== doc.uid);
```

Comment:
This is how I'm differentiating a student workspace document vs a my work or class work document, please let me know if there's a better solution.

Reply:
It depends on exactly what you're trying to achieve. This test will also be true for any `Class Work` documents published by users other than the current user. You could eliminate those by adding `&& !isPublishedDocument(doc)`. Alternatively, as you've already discovered, the documents shown in the student workspace section are the students' problem documents, so adding `&& doc.type === ProblemDocument` might be the best option.

Reply:
Yes I agree that adding `&& doc.type === ProblemDocument` would be best.

Reply:
I also think this logic should be extracted to `stores/ui.ts`. I'll add another comment about why.